### ISSUE TITLE
development version of pyinstaller installs all files named qt.conf of the original system

### ISSUE THREAD

**Issue by ejeandel** [pyinstaller/pyinstaller#4885]
Dear developers, 

There is a mistake in the current git version of pyinstaller, specifically caused by line 757 of this commit

https://github.com/pyinstaller/pyinstaller/commit/5d6715a24a51928658e38c26df596ca8ed7de386

On my linux version, and I expect most linuxes, the function collect_system_data in question will be called with path = "/usr"

By doing
```
    path = os.path.normpath(path)
    path = os.path.dirname(path)
```
The path will become "/". As a consequence, when pyinstaller will search for file qt.conf, it will search on the entire system, and not only in /usr, and will return all files named qt.conf it will find. This means it takes a long time and this is more seriously a  security problem, as my build directory now contains some private directory names on my system exposed by this mistake.

I assume the purpose of this line was not to transform "/usr" into "/"   :)

Thanks,

**Comment by Legorooj** [comment]
Damn. This needs verifying. @htgoebel could you do this?

**Comment by bjones1** [comment]
Do you have additional information to show that your analysis is correct? Or is this based on your examination of the code? You reference code in `collect_data_files`, not `collect_system_data`, so the diff you mention doesn't fit your description.

**Comment by ejeandel** [comment]
I basically put a `print(path)` before and after the two lines, and saw the "/usr" becoming "/"
Removing the second line (which was the line introduced by the patch) solved the problem. But I assume it's there for some reason related to Windows.


the beginning of the patch is about `collect_data_files`, but the line in question (line 757) is in `collect_system_data_files` (sorry for forgetting the `_files` part)

**Comment by bjones1** [comment]
Oops, sorry -- I didn't carefully read your post. I'll let @htgoebel comment on this, since it was his commit.

**Comment by bwoodsend** [comment]
I imagine this is also due to #5058

**Comment by Legorooj** [comment]
It is. I'll tag @rokm so he makes sure that this is fixed when fixing #5058.

**Comment by rokm** [comment]
This should be fixed now that #5110 has been merged.
